<launch>
  <arg name="debug_view" default="false" />
  <arg name="prefix" default="$(env HOME)/.ros/neatness_estimator" />

  <arg name="fe_node_name" default="objects_feature_extractor" />
  <arg name="de_node_name" default="distance_estimator" />
  <arg name="fe_service_topic" default="/$(arg fe_node_name)/extract" />
  <arg name="de_service_topic" default="/$(arg de_node_name)/estimate" />

  <arg name="cloud_topic" default="/openni_camera/depth_registered/points" />
  <arg name="image_topic" default="/openni_camera/rgb/image_rect_color" />
  <arg name="cluster_topic" default="/multi_euclidean_clustering/output_indices" />
  <arg name="instance_boxes_topic" default="/labeled_bounding_box_publisher/output/labeled_instance_boxes" />
  <arg name="cluster_boxes_topic" default="/labeled_bounding_box_publisher_aligned/output/labeled_cluster_boxes" />
  <arg name="labels_topic" default="/mask_rcnn_instance_segmentation/output/labels" />
  <arg name="classification_model_path" default="$(find neatness_estimator)/trained_data/motion_classification_model_191111.csv" />
  <arg name="geometry_feature" default="cvfh" />
  <arg name="motion_primitive_classifier" default="mlp" />

  <include file="$(find neatness_estimator)/launch/estimation_module_interface.launch">
    <arg name="input_cloud" default="$(arg cloud_topic)" />
    <arg name="input_image" default="$(arg image_topic)" />
    <arg name="input_cluster" default="$(arg cluster_topic)" />
    <arg name="input_instance_boxes" default="$(arg instance_boxes_topic)" />
    <arg name="input_cluster_boxes" default="$(arg cluster_boxes_topic)" />
    <arg name="input_labels" default="$(arg labels_topic)" />
    <arg name="prefix" default="$(arg prefix)" />
    <arg name="fe_service_topic" default="$(arg fe_service_topic)" />
    <arg name="de_service_topic" default="$(arg de_service_topic)" />
  </include>

  <include file="$(find neatness_estimator)/launch/objects_feature_extractor.launch">
    <arg name="fe_node_name" value="$(arg fe_node_name)" />
    <arg name="debug_view" default="false" />
    <arg name="sort_axis" default="0" />
    <arg name="save_reasoning_result" default="false" />
    <arg name="prefix" default="$(arg prefix)" />
    <arg name="cloud_topic" default="$(arg cloud_topic)" />
    <arg name="image_topic" default="$(arg image_topic)" />
    <arg name="cluster_topic" default="$(arg cluster_topic)" />
    <arg name="instance_boxes_topic" default="$(arg instance_boxes_topic)" />
    <arg name="cluster_boxes_topic" default="$(arg cluster_boxes_topic)" />
    <arg name="geometry_feature" default="$(arg geometry_feature)" />
  </include>

  <include file="$(find neatness_estimator)/launch/distance_estimator.launch" >
    <arg name="de_node_name" value="$(arg de_node_name)" />
    <arg name="prefix" value="$(arg prefix)" />
  </include>

  <include file="$(find neatness_estimator)/launch/get_motion_primitive.launch">
    <arg name="model_path" value="$(arg classification_model_path)" />
    <arg name="model" value="$(arg motion_primitive_classifier)" />
  </include>


  <include file="$(find neatness_estimator)/launch/data_corrector.launch">
    <arg name="model_path" value="$(arg classification_model_path)" />
  </include>

</launch>
